### ISSUE TITLE
Incorrect WER reported by nemo wer function

### ISSUE THREAD

**Issue by evilc3** [NVIDIA/NeMo#4919]
On my test set when I run  trainer.test() I am getting  a wer of 0.081. 

 global_step        │            0.0            │
│         test_loss         │     7.94022274017334      │
│         test_wer          │    0.08155332505702972  

but when I try running 
jiwer.wer or 
from nemo.collections.asr.metrics.wer import word_error_rate 

I am getting wer == 0.139 why is there such a big difference. 

have also tried : 

WER = WERBPE(decoding=ctc_bpe_decoding, use_cer = False,log_prediction=False, fold_consecutive=False,dist_sync_on_step=True)

Even this gives me a wer of 0.139

I am running this function for every sample and then taking mean 

```
from nemo.collections.asr.metrics.wer_bpe import CTCBPEDecoding,  WERBPE
from nemo.collections.asr.metrics.wer_bpe import CTCDecodingConfig

ctc_bpe_decoding  = CTCBPEDecoding(decoding_cfg = CTCDecodingConfig(), tokenizer = asr_model.tokenizer)

asr_model._wer.log_predictions  = False
asr_model._wer.use_cer  = False

WER = WERBPE(decoding=ctc_bpe_decoding, use_cer = False,log_prediction=False, fold_consecutive=False,dist_sync_on_step=True)

@torch.no_grad()
def cal_wer(gt, pred):
    global WER
    # print(asr_model.tokenizer.text_to_tokens(gt))

    gt = asr_model.tokenizer.text_to_ids(gt)
    pred = asr_model.tokenizer.text_to_ids(pred)

    len_gt = torch.tensor([len(gt)])
    len_pred = torch.tensor([len(pred)])
    
    gt = torch.tensor([gt])
    pred = torch.tensor([pred])

    WER.update(targets = gt, predictions = pred, target_lengths = len_gt, predictions_lengths=len_pred)

    return WER.compute()[0].item()
```

**Comment by titu1994** [comment]
How are you calculating wer across samples ? You should call WER only once providing a full list of all the predictions and ground truths, not one by one and then take average. 

Wer is a global metric - trainer.test() calculated the global wer properly. You can do it via https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_to_text_eval.py

**Comment by titu1994** [comment]
Wer, like F1 is a global metric. 8.1% here is correct cause it's the wer of the dataset. If you calculate wer per file, that's fine too, but it's then the wer of that particular file not the wer of the dataset.

**Comment by evilc3** [comment]
Got it thanks
